internal/component/loki/source/file/decompresser.go

```diff
@@ -82,7 +85,21 @@ func newDecompressor(
 
 	pos, err := positions.Get(path, labelsStr)
 	if err != nil {
-		return nil, fmt.Errorf("failed to get positions: %w", err)
+		level.Error(logger).Log("msg", "failed to get file position", "file", path, "error", err)
+		switch onPositionsFileError {
+		case OnPositionsFileErrorSkip:
+			return nil, fmt.Errorf("failed to get file position: %w", err)
+		case OnPositionsFileErrorRestartEnd:
+			level.Warn(logger).Log("msg", "`restart_from_end` is not supported for compressed files, defaulting to `skip`")
+			return nil, fmt.Errorf("failed to get file position: %w", err)
+		case OnPositionsFileErrorRestartStart:
+			pos = 0
+			positions.Put(path, labelsStr, pos)
+			level.Info(logger).Log("msg", "reset position to start of file after positions error")
+		default:
+			level.Debug(logger).Log("msg", "unrecognized `on_positions_file_error` option, defaulting to `skip`", "option", onPositionsFileError)
+			return nil, fmt.Errorf("failed to get file position: %w", err)
+		}
```

Comment:
This is identical in decompresser and tailer, maybe a helper function?

Reply:
It's _almost_ identical, decompressor doesn't support `restart_from_end`. I thought about decomposing but didn't like my ideas for adding specific behavior for that behavior being different per use.